crates/proto/src/op/message.rs

```diff
@@ -1313,4 +1305,238 @@ mod tests {
         let encoded = message.to_bytes().unwrap();
         Message::from_bytes(&encoded).expect("failed to parse encoded message");
     }
+
+    #[test]
+    #[cfg(feature = "std")]
+    fn test_read_records() {
```

Comment:
I think this would be better if each "test block" was a separate test? I don't think there's a lot of shared state that make a single test function much more ergonomic?

Reply:
Done. I also reworked some to avoid std dependence where possible.